src/text_info.rs

```diff
@@ -8,15 +8,31 @@ use super::types::LineAndColumnIndex;
 use crate::swc::common::BytePos;
 use crate::swc::common::Span;
 
+/// Stores the source text along with other data such as where all the lines
+/// occur in the text.
+///
+/// Note: This struct is cheap to clone.
 #[derive(Clone)]
 pub struct SourceTextInfo {
+  // keep this struct cheap to clone
   start_pos: BytePos,
   text: Arc<String>,
   text_lines: Arc<TextLines>,
 }
 
 impl SourceTextInfo {
-  pub fn new(start_pos: BytePos, text: Arc<String>) -> Self {
+  /// Creates a new `SourceTextInfo` from the provided source text.
+  pub fn new(text: Arc<String>) -> Self {
```

Comment:
I change the signatures because this is the most common case. Will get rid of the `BytePos(0)` stuff going on mostly everywhere.